src/lib.rs

```diff
@@ -51,6 +51,42 @@ lazy_static! {
     static ref PATCHED_IMAGE_DOCKER_TAG: OsString = OsString::from_str("mx-tester/synapse").unwrap();
 }
 
+#[derive(Debug, Deserialize)]
+pub struct DockerConfig {
+    #[serde(default)]
+    /// A docker network to run the synape containers on.
+    /// If the network does not exist it will be created.
+    /// If a network is provided, the synapse container will be given the hostname `synapse`.
+    pub docker_network: Option<String>,
+
+    /// The hostname to give the synapse container on the docker network, if the docker network has been provided.
+    #[serde(default = "DockerConfig::hostname_default")]
+    pub hostname: String,
+
+    #[serde(default = "DockerConfig::port_mapping_default")]
+    /// The docker port mapping configuration to use for the synapse container e.g. `9999:9999`.
+    pub port_mapping: String,
+}
+
+impl DockerConfig {
+    fn hostname_default() -> String {
+        "synapse".to_string()
+    }
+    fn port_mapping_default() -> String {
+        "9999:9999".to_string()
+    }
+}
+
+impl Default for DockerConfig {
```

Comment:
Implementing both `Default` and `#[serde(default = ...)]` feels like a code smell. I think you should get rid of the `#[serde(default = ...)]` and inline `hostname_default`, `port_mapping_default` in `fn default()`.